src/node_api.cc

```diff
@@ -627,26 +692,19 @@ NAPI_NO_RETURN void napi_fatal_error(const char* location,
 }
 
 napi_status napi_open_callback_scope(napi_env env,
-                                     napi_value resource_object,
+                                     napi_value /** ignored */,
```

Comment:
Should we perhaps use the `resource_object` to warn if it's different than the object contained in the `AsyncContext`?

Reply:
@quad I see what you mean, but that might be a breaking change and we don't have breaking changes for N-API methods.  If we felt strongly about it we should add a new method without the parameter and doc deprecate the existing method (but not remove of course). I think we did something similar for another API in terms of indicate /** ignored */

Reply:
I was just thinking a log message warning, no change in behaviour. Is that still considered breaking? Either way, not blocking on that. Just a thought that it'd be nice to have some indicator when the API is used improperly.

Reply:
To my knowledge node has no logger. Logging to stdout/stderr tends to break command line tools.

But I think we should find a way how to deprecate something in NAPI in a way more visible to users like comments in doc.
I think a compile time warning like it is used for [node::MakeCallback](https://github.com/nodejs/node/blob/936ce85c0be81f73304249aa5861436d4b71cc23/src/node.h#L176) would be great.

Reply:
> I think a compile time warning like it is used for node::MakeCallback would be great.

AFAICT, it is not possible to emit compile-time warning without changing the API shapes. i.e. the signature of node::MakeCallback was migrated to the one with an additional parameter async_context. Though in the case of n_api, the deprecation is the parameter in the middle of the parameter list. Please correct me if I got a wrong impression.